libp2p/peer/id.py

```diff
@@ -64,21 +54,14 @@ def from_base58(cls, b58_encoded_peer_id_str: str) -> "ID":
         return pid
 
     @classmethod
-    def from_pubkey(cls, key: RsaKey) -> "ID":
-        # export into binary format
-        key_bin = key.exportKey("DER")
-
-        algo: int = multihash.Func.sha2_256
-        # TODO: seems identity is not yet supported in pymultihash
-        # if len(b) <= MAX_INLINE_KEY_LENGTH:
-        #     algo multihash.func.identity
-
-        mh_digest: multihash.Multihash = multihash.digest(key_bin, algo)
+    def from_pubkey(cls, key: bytes) -> "ID":
+        algo = multihash.Func.sha2_256
+        mh_digest = multihash.digest(key, algo)
         return cls(mh_digest.encode())
 
     @classmethod
-    def from_privkey(cls, key: RsaKey) -> "ID":
-        return cls.from_pubkey(key.publickey())
+    def from_privkey(cls, public_key: bytes) -> "ID":
```

Comment:
This should be `private_key`. But then it can't be type `bytes`...🤔

Reply:
yeah -- i actually had more to add here from another branch!

i ended up just dropping the `from_privkey` method, i don't see how we can keep it generic to a particular cryptosystem while taking private key bytes (bc it will be specific to the algo to get the public key)